core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveDeltaFloatingCodec.java

```diff
@@ -255,6 +255,12 @@ public void decodeAndFillVector(byte[] pageData, ColumnVectorInfo vectorInfo, Bi
       CarbonColumnVector vector = vectorInfo.vector;
       BitSet deletedRows = vectorInfo.deletedRows;
       DataType vectorDataType = vector.getType();
+      VectorUtil vectorUtil = new VectorUtil(vectorInfo, pageSize, vector, vectorDataType)
```

Comment:
can we handle this in called method itself, handling in each and class its not correct, pls change this in other classes too

Reply:
Direct compress needed a different handling because of decimal. So, had to update in each

Reply:
Also, decodeAndFillVector takes vectorInfo, not the vectors. So, all interface need to change if I do this.

Reply:
removed this whole class itself and updating vector inside ColumnarVectorWrapperDirectFactory
.getDirectVectorWrapperFactory